### ISSUE TITLE
Clarification on 'Beta' Calculation in Eigenpro3 Functions

### ISSUE THREAD

**Issue by phanatchakrit** [parthe/torchkernels#2]
While reviewing the code for the nystrom_kernel_svd in Eigenpro3 `beta = torch.from_numpy(kmat).diag().max()` and top_eigensystem `beta = n * (scaled_kmat.diag() - (E[:,:q].pow(2)*(L[:q]-L[q])).sum(-1)).max()` functions, Could you please clarify the appropriate use cases and considerations that led to the selection of each 'beta' calculation method in these functions? I also observed that 'beta' in nystrom_kernel_svd always equals 1.0, whereas in top_eigensystem, it does not.

**Comment by parthe** [comment]
'beta' is defined as max of K(xi,xi), over all samples xi in training data.
This is an upper bound for the Lipschitz constant of the loss per sample.

K(xi,xi) = norm(K(xi,.))**2 = trace(Hessian_f(f(xi)-yi)**2)) where the norm is the RKHS norm and the Hessian is wrt the RKHS inner product (using Frechet derivatives).

See the paper for dependence of learning rate and optimal batch size on this quantity: https://arxiv.org/abs/1712.06559

But when we precondition using a spectral preconditioner (as in https://arxiv.org/abs/1806.06144) then the largest Lipschitz constant can be shown to be lower).
